language/move-lang/src/hlir/translate.rs

```diff
@@ -987,7 +987,29 @@ fn exp_impl(context: &mut Context, result: &mut Block, e: T::Exp) -> H::Exp {
             };
             HE::BorrowLocal(mut_, tmp)
         }
-
+        TE::Cast(te, rhs_ty) => {
+            use H::BaseType_ as B;
+            use H::SingleType_ as S;
+            use H::Type_ as T;
+            use N::BuiltinTypeName_ as BT;
+            use N::TypeName_ as TN;
+            macro_rules! builtin {
```

Comment:
This is pretty fancy (i.e., hard to follow). I haven't done anything with this hlir phase before, but I'm surprised that we don't already have some code to translate these builtin types. I guess this is fine for now, but you might consider splitting this into a separate function or adding some comments to explain what is going on here and why it is done this way.

Reply:
I'll make a function thats less confusing